common/djangoapps/microsite_configuration/backends/database.py

```diff
@@ -32,6 +35,9 @@ def set_config_by_domain(self, domain):
         if not self.has_configuration_set() or not domain:
             return
 
+        # CDODGE: I'm concerned about this performance. Seems like we should cache this mapping
+        # and maybe auto expire the cache extry after 5 minutes. This method will be called on
+        # every request into edx-platform
```

Comment:
I agree with this. I though about letting the database handle it by doing.filter(subdomian__startswith=domain) but the results are not the same. Cache looks like a good idea here